File: solutions/python/Maximum Depth of Binary Tree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root==None:
            return 0;
        queue=[root]
        depth=0
        while queue:
            size=len(queue)
            i=0
            while i< size:
                node=queue.pop(0)
                if node.left:

                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
                i=i+1
            depth=depth+1
        return depth

File: solutions/cpp/Maximum Depth of Binary Tree.cpp
#include <iostream>
#include <queue>
struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
public:
    int maxDepth(TreeNode *root)
    {
        if (root == NULL)
            return 0;
        std::queue<TreeNode *> q;
        int depth = 0;
        q.push(root);
        while (!q.empty())
        {
            int size = q.size(), i = 0;
            while (i < size)
            {
                TreeNode *node = q.front();
                q.pop();
                if (node->left)
                    q.push(node->left);
                if (node->right)
                    q.push(node->right);
                i = i + 1;
            }
            depth = depth + 1;
        }
        return depth;
    }
};